src/main/scala/org/dia/algorithms/mcc/GTGRunner.scala

```diff
@@ -33,24 +33,50 @@ import org.dia.core.{SciSparkContext, SciTensor}
  */
 class GTGRunner(val masterURL: String,
                 val paths: String,
-                val partitions: Int) {
+                val partitions: Int,
+                val maxAreaOverlapThreshold : Double = 0.65,
+                val minAreaOverlapThreshold : Double = 0.50,
+                val outerTemp : Double = 241.0,
+                val innerTemp : Double = 233.0,
+                val convectiveFraction : Double = 0.9,
+                val minArea : Int = 625,
+                val nodeMinArea : Int = 150) {
 
   val logger = org.slf4j.LoggerFactory.getLogger(this.getClass)
 
   /**
+   * To create a map of Nodes from the edges found.
+   *
+   * @param edges sequence of MCCEdge objects
+   * @param lat the lattitude dimension array
+   * @param lon the longitude dimension array
+   * @return
+   */
+  def createNodeMapFromEdgeList(edges: Seq[MCCEdge],
+                                lat: Array[Double],
+                                lon: Array[Double]): mutable.HashMap[String, MCCNode] = {
+
+    val MCCNodes = edges.flatMap(edge => List(edge.srcNode, edge.destNode)).distinct
+    val MCCNodeKeyValuesSet = MCCNodes.map(node => {
+      val key = s"${node.frameNum},${node.cloudElemNum}"
+      node.updateLatLon(lat, lon)
+      (key, node)
+    })
+    mutable.HashMap[String, MCCNode](MCCNodeKeyValuesSet: _*)
+  }
+
+  /**
    * Records the frame number in all SciTensors stored in the RDD
    * Preconditon : The files read are are of the form merg_XX_4km-pixel.nc
    *
    * @param sRDD input RDD of SciTensors
    * @return RDD of SciTensors with Frame number recorded in metadaa table
    */
-  def recordFrameNumber(sRDD: RDD[SciTensor]): RDD[SciTensor] = {
+  def recordFrameNumber(sRDD: RDD[SciDataset]): RDD[SciDataset] = {
```

Comment:
Kim great question!
This goes back to issue #67

I think we should just keep it as an RDD of SciDatasets. If you look at Sparks context code they too return multiple types of rdd's like hadooprdd, binaryrdd, etc. However, the return type is always a normal rdd (since the others are all subclasses of the normal rdd). The different types of rdd's encapsulate the different sources they're loading from. 

There's a couple more challenges I outlined in the original issue. 

@kwhitehall how do you feel about this? 
We can coin an RDD of SciDatasets as an SRDD.
It doesn't have to have its own class.
Furthermore all the functions can be implicitly called on these types of RDDs by importing SRDDFunctions